.github/workflows/build.yaml

```diff
@@ -79,6 +80,6 @@ jobs:
         run: tar -xvf ttm_${{ matrix.arch }}-${{ matrix.config}}.tar
       - name: Build tt-metal tests
         run: |
-          rm -f build/CMakeCache.txt
-          cmake -B build -G Ninja
-          cmake --build build --target tests -- -j`nproc`
+          rm -f build/CMakeCache.txt && find build -name 'cmake_pch.hxx.pch' -exec touch {} \;
+          cmake -B build -G Ninja -DCMAKE_CXX_COMPILER=clang++-17
```

Comment:
I think it is better to specify cxx version in the cmake file itself

# Set the C++ compiler
```
set(CMAKE_CXX_COMPILER "clang++-17")

# Other project settings
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

or maybe even start using a toolchain file, preparing to package management onboarding
`CMAKE_TOOLCHAIN_FILE`

Reply:
If we set it, we won't be able to override clang17 through command line

I agree we should look into a toolchain file in the future.